File: goblinframework-core/src/main/java/org/goblinframework/core/monitor/FlightEvent.java
package org.goblinframework.core.monitor;

import org.goblinframework.api.event.GoblinEvent;
import org.goblinframework.api.event.GoblinEventChannel;
import org.goblinframework.api.monitor.Flight;
import org.jetbrains.annotations.NotNull;

@GoblinEventChannel("/goblin/monitor")
public class FlightEvent extends GoblinEvent {

  private final Flight flight;

  public FlightEvent(@NotNull Flight flight) {
    this.flight = flight;
  }

  @NotNull
  public Flight getFlight() {
    return flight;
  }
}

File: goblinframework-monitor/src/main/java/org/goblinframework/monitor/flight/FlightEvent.kt
package org.goblinframework.monitor.flight

import org.goblinframework.core.event.GoblinEvent
import org.goblinframework.core.event.GoblinEventChannel

@GoblinEventChannel("/goblin/monitor")
class FlightEvent(val flight: Flight) : GoblinEvent() {

  init {
    isFair = true
  }

}
